### PR TITLE
feat: Add CustomArrayList and CustomHashSet implementations

### PR DESCRIPTION
**Pull request by tomrist9**
Pull Request Description
This pull request introduces the following features:

CustomArrayList
A custom implementation of ArrayList with the following functionalities:
Add (put) elements to the list.
Retrieve elements by index using the get method.
Delete elements by index using the delete method.
Automatically resizes when the internal capacity is exceeded.
Includes an overridden toString() method for better debugging and output representation.
CustomHashSet
A custom implementation of HashSet that ensures object uniqueness and provides:
Add (put) unique elements.
Check if an element exists using the contains method.
Delete elements using the delete method.
Iterate over elements using a custom iterator and support for while and for-each loops.
Automatic resizing when the load factor exceeds the predefined threshold.
Main Class
Added a Main class to test the implemented functionalities:
Validates the behavior of CustomArrayList and CustomHashSet.
Demonstrates adding, retrieving, deleting, resizing, and iterating elements.

### CODE REVIEW

File: andersen-homework/src/main/java/org/util/CustomArrayList.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package org.util;
+
+import java.util.Arrays;
+
+public class CustomArrayList <T>{
+    private Object[] items;
+    int size=0;
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/util/CustomArrayList.java]: What this field has default access modifier?

File: andersen-homework/src/main/java/org/util/CustomArrayList.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public CustomArrayList() {
+        items = new Object[10];
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/util/CustomArrayList.java]: put please 10 as a constant

File: andersen-homework/src/main/java/org/util/CustomArrayList.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package org.util;
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/util/CustomArrayList.java:1]: need to apply formatting for the class

File: andersen-homework/src/main/java/org/util/CustomArrayList.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            items[i] = items[i + 1];
+        }
+        items[--size] = null;
+    }
+    private void resize(){
+        items= Arrays.copyOf(items, items.length*2);
+    }
+
+
+    public int getSize() {
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/util/CustomArrayList.java:45]: Why do we need 2 methods with the same body?

File: andersen-homework/src/main/java/org/util/CustomHashSet.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
+package org.util;
+
+import java.util.LinkedList;
+
+public class CustomHashSet<T> {
```
Comment (Darya-Drutsko) [inline andersen-homework/src/main/java/org/util/CustomHashSet.java:5]: Cool implementation!

**Review by Darya-Drutsko — COMMENTED** [review]
No need to delete previous ones files, lets have them separate(e.g. in different packages), but in one pr

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 1 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java]: int size has a default access modifier.It makes field accessible within the same package

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 2 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java]: done

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 4 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java:45]: I used getter firstly, but could not call it in main class. so created size class manually. But then I removed getter annotation and created getter manually.

Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java:45]: please review new commit after i made changes

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 2 above)
Comment (Darya-Drutsko) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java]: Don't see the result

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 3 above)
Comment (Darya-Drutsko) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java:1]: Don't see the result

**Review by Darya-Drutsko — CHANGES_REQUESTED** [review]

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 3 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java:1]: done

File: andersen-homework/src/main/java/org/util/CustomArrayList.java (hunk 2 above)
Comment (tomrist9) [reply andersen-homework/src/main/java/org/util/CustomArrayList.java]: done

**Review by Darya-Drutsko — COMMENTED** [review]
Always gitignore .idea folder and return deleted files, they should still be present in develop branch after merging

**Pull request merged**